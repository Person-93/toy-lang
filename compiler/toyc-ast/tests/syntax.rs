use anyhow::Result;
use chumsky::Parser;
use test_util::{assert_debug_snapshot, snapshots};
use toyc_ast::{ast, tokens::TokenIter};

fn main() -> Result<()> {
  snapshots(|text| {
    let tokens = TokenIter::new(text).into_stream();
    assert_debug_snapshot!(ast::parse::file().parse_recovery(tokens));
  })
}
