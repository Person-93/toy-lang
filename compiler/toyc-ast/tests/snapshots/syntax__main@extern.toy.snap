---
source: compiler/toyc-ast/tests/syntax.rs
assertion_line: 8
expression: "ast::parse::file().parse_recovery(tokens)"

---
(
    Some(
        File {
            attrs: [],
            items: [
                Item {
                    attrs: [],
                    vis: None,
                    item_kind: Function(
                        Function {
                            extern_: Some(
                                "C",
                            ),
                            ident: answer @ 14..20,
                            generic_params: None,
                            params: [],
                            return_type: Some(
                                Ident(
                                    i8 @ 26..28,
                                ),
                            ),
                            body: Some(
                                Expr {
                                    primary: CodeBlock(
                                        CodeBlock {
                                            statements: [],
                                            trailing: Some(
                                                Expr {
                                                    primary: NumLit(
                                                        NumLit {
                                                            prefix: None,
                                                            val: 42,
                                                            decimal: None,
                                                            ty: None,
                                                            span: Span {
                                                                lo: 32,
                                                                hi: 34,
                                                            },
                                                        },
                                                    ),
                                                    args: None,
                                                    span: Span {
                                                        lo: 32,
                                                        hi: 34,
                                                    },
                                                },
                                            ),
                                            span: Span {
                                                lo: 32,
                                                hi: 34,
                                            },
                                        },
                                    ),
                                    args: None,
                                    span: Span {
                                        lo: 30,
                                        hi: 36,
                                    },
                                },
                            ),
                            span: Span {
                                lo: 0,
                                hi: 36,
                            },
                        },
                    ),
                    span: Span {
                        lo: 0,
                        hi: 36,
                    },
                },
                Item {
                    attrs: [],
                    vis: None,
                    item_kind: Function(
                        Function {
                            extern_: Some(
                                "C",
                            ),
                            ident: ping @ 52..56,
                            generic_params: None,
                            params: [],
                            return_type: Some(
                                Ident(
                                    usize @ 62..67,
                                ),
                            ),
                            body: Some(
                                Expr {
                                    primary: CodeBlock(
                                        CodeBlock {
                                            statements: [],
                                            trailing: Some(
                                                Expr {
                                                    primary: Ident(
                                                        ping @ 71..75,
                                                    ),
                                                    args: Some(
                                                        [],
                                                    ),
                                                    span: Span {
                                                        lo: 71,
                                                        hi: 77,
                                                    },
                                                },
                                            ),
                                            span: Span {
                                                lo: 71,
                                                hi: 77,
                                            },
                                        },
                                    ),
                                    args: None,
                                    span: Span {
                                        lo: 69,
                                        hi: 79,
                                    },
                                },
                            ),
                            span: Span {
                                lo: 38,
                                hi: 79,
                            },
                        },
                    ),
                    span: Span {
                        lo: 38,
                        hi: 79,
                    },
                },
                Item {
                    attrs: [],
                    vis: None,
                    item_kind: Function(
                        Function {
                            extern_: Some(
                                "C",
                            ),
                            ident: pong @ 95..99,
                            generic_params: None,
                            params: [],
                            return_type: Some(
                                Ident(
                                    usize @ 105..110,
                                ),
                            ),
                            body: None,
                            span: Span {
                                lo: 81,
                                hi: 111,
                            },
                        },
                    ),
                    span: Span {
                        lo: 81,
                        hi: 111,
                    },
                },
            ],
            span: Span {
                lo: 0,
                hi: 111,
            },
        },
    ),
    [],
)
